graphql_compiler/cost_estimation/statistics.py

```diff
@@ -1,10 +1,28 @@
 # Copyright 2019-present Kensho Technologies, LLC.
 from abc import ABCMeta, abstractmethod
+from dataclasses import dataclass
 import datetime
+import math
+from typing import Any, Dict, List, Optional, Tuple
 
 import six
 
 
+@dataclass
+class SamplingSummary:
+    """Results extracted from sampling a random rowset that are relevant for statistics."""
+
+    # The class this summary describes
+    class_name: str
+
+    # Mapping field_name -> field_value -> observed_count for all field names and observed
+    # field values on this class.
+    value_counts: Dict[str, Dict[Any, int]]
+
+    # The number_of_rows / number_of_samples ratio
+    sample_ratio: int
```

Comment:
Maybe record the number of samples as well? I know it can be calculated by summing over the counts in `value_counts.values()` as well, but if there are a lot of them, it might be a more pleasant API to just have it ready.

Reply:
Hmm, could be useful to sanity-check `sample_ratio * number_of_samples ~= row_count`. I'd leave it for a future PR though, if you don't mind. There's many things to do for statistics validation, like canonicalizing values, and this can be done as part of that issue